pages/accountLists/[accountListId]/reports/salaryCurrency.page.tsx

```diff
@@ -31,4 +44,18 @@ const SalaryReportPage = (): ReactElement => {
   );
 };
 
+export const getServerSideProps: GetServerSideProps = async ({ req, res }) => {
+  const session = await getSession({ req });
+
+  if (!session?.user['token']) {
+    res.writeHead(302, { Location: '/' });
+    res.end();
+    return { props: {} };
+  }
```

Comment:
@OzzieOrca That's a good point. We could build a hook named `useAuth` which connects and manages contexts specifically using JWT token for user auth. And also need to wrap the entire app with the Auth Provider.

I will look into the Onekey auth.

Is there any specific reason to use Onekey for the user authentication?

Reply:
I'm not sure what you mean by "Onekey".

We're using The Key (based on CAS) currently which is an internal SSO system. As an organization we are migrating to Okta so that change might be coming soon.

Ya getting this somewhere globally so we don't have to have auth redirect code in every page would be great.

Reply:
@OzzieOrca Sorry, that's my bad. I was meaning "The Key".
Let me work on the auth stuff.

Reply:
I wonder if we shouldn't just skip The Key in the React version and go straight to Okta. I think we will migrate to Okta before the rewrite is done.

Reply:
That would be great.